tests/apiserver-proxy-tests/apiserver-proxy/internal/config.go

```diff
@@ -0,0 +1,80 @@
+package graphql
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/vrischmann/envconfig"
+)
+
+type User string
+
+const (
+	AdminUser    User = "admin"
+	ReadOnlyUser User = "read-only"
+	NoRightsUser User = "no-rights"
+	NoUser       User = "no-user"
+)
+
+type envConfig struct {
+	Domain        string `envconfig:"default=kyma.local"`
+	AdminEmail    string
+	AdminPassword string
+}
+
+type config struct {
+	GraphQLEndpoint  string
+	IdProviderConfig idProviderConfig
+	EnvConfig        envConfig
+}
+
+type idProviderConfig struct {
+	DexConfig       dexConfig
+	ClientConfig    clientConfig
+	UserCredentials userCredentials
+}
+
+type dexConfig struct {
+	BaseUrl           string
+	AuthorizeEndpoint string
+	TokenEndpoint     string
+}
+
+type clientConfig struct {
+	ID          string
+	RedirectUri string
+}
+
+type userCredentials struct {
+	Username string
+	Password string
+}
+
+func LoadConfig(userContext User) (config, error) {
```

Comment:
You return unexported type from exported function. Caller won't be able to do anything meaningful with it. Make `config` public.